web/src/test/java/org/springframework/security/web/server/csrf/CsrfWebFilterTests.java

```diff
@@ -183,6 +188,22 @@ public void filterWhenPostAndEstablishedCsrfTokenAndHeaderValidTokenThenContinue
 		chainResult.assertWasSubscribed();
 	}
 
+	@Test
+	public void filterWhenNonStandardHTTPMethodUsedThenMatchesRequireCSRFProtection() {
+		MockServerWebExchange nonStandardHttpRequest = from(MockNonStandardHttpRequest.nonStandardHTTPMethod("/"));
+
+		ServerWebExchangeMatcher serverWebExchangeMatcher = spy(CsrfWebFilter.DEFAULT_CSRF_MATCHER);
```

Comment:
I don't think we need to test the `CsrfWebFilter` and the matcher here. I think we can just check `CsrfWebFilter.DEFAULT_CSRF_MATCHER` directly. If we need to test both, we should probably create separate tests.